sphinxgallery/gen_rst.py

```diff
@@ -444,7 +458,12 @@ def generate_file_rst(fname, target_dir, src_dir):
     ref_fname = example_file.replace(os.path.sep, '_')
     example_rst = """\n\n.. _sphx_glr_{0}:\n\n""".format(ref_fname)
 
-    if not fname.startswith('plot'):
+    exec_script = fname.startswith('plot')
+    use_mayavi = gallery_conf.get('use_mayavi', False)
+    if not use_mayavi and any('mayavi' in bl[1] for bl in script_blocks):
```

Comment:
I am not sure I understand the logic here, if use_mayavi is False and mayavi is used by the script, why should there not be an error? Basically is there a use case for not showing mayavi figures even if there are some generated by the example?

This also begs the question, why do we need `use_mayavi` in the first place? Couldn't we try to import mayavi only when needed i.e. if I understand correctly when looking for mayavi figures in gen_rst.save_figures?